### ISSUE TITLE
detection tutorial raises ValueError("cls_logits is NaN or Inf.")

### ISSUE THREAD

**Issue by taborzbislaw** [Project-MONAI/tutorials#1122]
Hi,

I am running detection tutorial on my dataset (RM examination of Liver, four input channels). I have prepared dataset analogously to the LUNA16 dataset used in the tutorial (the only difference is that in LUNA16 dataset half width, height and depth are all equal while for my dataset they are all different). I slightly modified config_train_liver.json to match it to the properties of my dataset (I decreased patch size to fit it to the minimum shape of the input images, set input_channels to 4, and set spacing in accordance with the affinity matrix of input images, anchor shapes are left unchanged). 
I also slightly changed the implementation of generate_detection_train_transform: the intensity_transform is replaced by identity transform (my input images are already scaled to (0,1) range), and rotations are removed.

The training starts correctly but after some steps I got an error:

111/207, train_loss: 0.5020
112/207, train_loss: 1.0010
113/207, train_loss: 0.9321
114/207, train_loss: 0.2274
115/207, train_loss: 0.7485
116/207, train_loss: 1.0342
117/207, train_loss: 1.0312
118/207, train_loss: 0.8242
119/207, train_loss: 1.0352
120/207, train_loss: 0.8623
121/207, train_loss: 1.0400
122/207, train_loss: 1.0742
123/207, train_loss: 0.9609
124/207, train_loss: 0.8711
Traceback (most recent call last):
  File "luna16_training.py", line 457, in <module>
    main()
  File "luna16_training.py", line 294, in main
    outputs = detector(inputs, targets)
  File "/home/zbislaw_tabor_onwelo_com/monai/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1190, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/zbislaw_tabor_onwelo_com/monai/lib/python3.8/site-packages/monai/apps/detection/networks/retinanet_detector.py", line 488, in forward
    head_outputs = self.network(images)
  File "/home/zbislaw_tabor_onwelo_com/monai/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1190, in _call_impl
    return forward_call(*input, **kwargs)
torch.jit.Error: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript (most recent call last):
  File "/home/zbislaw_tabor_onwelo_com/monai/lib/python3.8/site-packages/monai/apps/detection/networks/retinanet_network.py", line 125, in forward
    
            if torch.isnan(cls_logits).any() or torch.isinf(cls_logits).any():
                raise ValueError("cls_logits is NaN or Inf.")
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
    
        return cls_logits_maps
builtins.ValueError: cls_logits is NaN or Inf.


Could I ask for help in solving this problem?

**Comment by stephahn** [comment]
Hi @taborzbislaw  I face exactly the same problem. Did you find a solution for the Nan or Inf output?
thank you in advance

**Comment by taborzbislaw** [comment]
Hi. No I have not solved the problem. The only VI neural network framework which I have found and which is certainly working is tensorflow-probability: https://www.tensorflow.org/probability?hl=en
I have however tested it only for classification networks but not for segmentation networks. For segmentation networks I have used MC Dropout but this is quite trivial (except selection of the dropout rate).
